numba/cpython/setobj.py

```diff
@@ -947,11 +947,13 @@ def _resize(self, payload, nentries, errmsg):
                                               (errmsg,))
 
         # Re-insert old entries
+        # Decref to balance incref from `_add_key`
```

Comment:
Although I don't think it would ever be an issue because the only time there's the incref/decref combo would be when copying the data from one set to another where the item should always be `not_found`, I do think having a kwarg makes the code simpler and more understandable

Reply:
I agree, in this specific case it shouldn't be a problem by the nature of its construction, but in general I think would be. Thanks for adding to the API, seems like it'd make sense to be able to toggle it at the call site.